akka-actor-typed/src/main/scala/akka/actor/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -202,3 +211,38 @@ private final class BuiltBehavior[T](messageHandlers: Array[Case[T, T]], signalH
     }
   }
 }
+
+/**
+ * The concrete behavior
+ *
+ * INTERNAL API
+ */
+@InternalApi
+private final class BuiltClassifiedBehavior[T](
+    classifiedMessageBehaviors: Map[Class[_ <: T], Array[Case[T, T]]],
+    signalHandlers: Array[Case[T, Signal]])
+    extends ExtensibleBehavior[T] {
+
+  override def receive(ctx: TypedActorContext[T], msg: T): Behavior[T] =
+    classifiedMessageBehaviors.get(msg.getClass) match {
+      case Some(handlers) => receive(msg, handlers, 0)
```

Comment:
When looking at this I'm not convinced that it's worth adding this optimization. The map lookup vs traversing the array?

Reply:
@He-Pin shall we close this attempt?